website/docs/r/vapp_vm.html.markdown

```diff
@@ -97,24 +119,59 @@ The following arguments are supported:
-* `disk` - (Optional; *v2.1+*) Independent disk attachment configuration. Details below
+* `disk` - (Optional; *v2.1+*) Independent disk attachment configuration. See [Disk](#disk) below for details.
 * `expose_hardware_virtualization` - (Optional; *v2.2+*) Boolean for exposing full CPU virtualization to the
 guest operating system so that applications that require hardware virtualization can run on virtual machines without binary
 translation or paravirtualization. Useful for hypervisor nesting provided underlying hardware supports it. Default is `false`.
+* `network` - (Optional; *v2.2+*) A block to define network interface. Multiple can be used. See [Network](#network) and 
+example for usage details. **Deprecates**: `network_name`, `ip`, `vapp_network_name`. **Note**: this property and all
+its parameters do force recreation of VMs!
 
-Independent disk support the following attributes:
+<a id="disk"></a>
+## Disk
 
 * `name` - (Required) Independent disk name
 * `bus_number` - (Required) Bus number on which to place the disk controller
 * `unit_number` - (Required) Unit number (slot) on the bus specified by BusNumber.
+
+
+<a id="network"></a>
+## Network
+
+* `type` (Required) Network type, one of: `none`, `vapp` or `vdc`. `none` creates a NIC with no network attached, `vapp` attaches a vApp network, while `vdc` attaches organization VDC network.
+* `name` (Optional) Name of the network this VM should connect to. Always required except for `type` `NONE`.
+* `is_primary` (Optional) Set to true if network interface should be primary. First network card in the list will be primary by default.
+* `mac` - (Computed) Mac address of network interface.
+* `ip_allocation_mode` (Required) IP address allocation mode. One of `POOL`, `DHCP`, `MANUAL`, `NONE`:  
+
+  * `POOL` - Static IP address is allocated automatically from defined static pool in network.
+  
+  * `DHCP` - IP address is obtained from a DHCP service. Field `ip` is not guaranteed to be populated. Because of this it may appear
+  after multiple `terraform refresh` operations.
+  
+  * `MANUAL` - IP address is assigned manually in the `ip` field. Must be valid IP address from static pool.
+  
+  * `NONE` - No IP address will be set because VM will have a NIC without network.
+
+* `ip` (Optional, Computed) Settings depend on `ip_allocation_mode`. Field requirements for each `ip_allocation_mode` are listed below:
+
+  * `ip_allocation_mode=POOL` - **`ip`** value must be omitted or empty string "". Empty string may be useful when doing HCL
+  variable interpolation. Field `ip` will be populated with an assigned IP from static pool after run.
+  
+  * `ip_allocation_mode=DHCP` - **`ip`** value must be omitted or empty string "". Field `ip` is not guaranteed to be populated
+  after run due to the VM may not work properly with DHCP or may be missing VMware tools. Because of this `ip` may also appear after multiple `terraform refresh` operations when is reported back to vCD.
```

Comment:
"due to the VM may not work properly with DHCP or may be missing VMware tools" ->
"due to the VM lacking VMware tools or not working properly with DHCP"

Reply:
done.